lib/providers/filesystem/index.js

```diff
@@ -152,6 +153,7 @@ FileSystemProvider.prototype.getContainer = function(containerName, cb) {
   var self = this;
   if (!validateName(containerName, cb)) return;
   var dir = path.join(this.root, containerName);
+  this.getProperPath(dir, this.root);
```

Comment:
We already have a method to detect invalid container and file names, see `validateName`. Please fix this existing method instead of adding a new one.